capplets/desktop-icons/mate-di-capplet.c

```diff
@@ -0,0 +1,380 @@
+		gtk_button_set_label(GTK_BUTTON(button), "gtk-edit");
+	}
+}
+
+/* Leave edit icon name mode and save new icon name for trash icon */
+void
+on_activate_entry_trash_cb(GtkWidget *entry, MateDICapplet* capplet)
+{
+    const gchar *icon_name = gtk_entry_get_text(GTK_ENTRY(entry));
+
+	g_settings_set_string(capplet->desktop_icons_settings, TRASH_NAME_KEY, icon_name);
+ 
+    gtk_button_set_label(GTK_BUTTON(capplet->button_trash_toggle), icon_name);
+
+	gtk_widget_show(capplet->button_trash_toggle);
+	gtk_widget_hide(capplet->entry_trash);
+	gtk_button_set_label(GTK_BUTTON(capplet->button_edit_trash), "gtk-edit");
+}
+
+/* Enter edit icon name for network icon */
+static void
+edit_network_icon_name_cb(GtkWidget* button, MateDICapplet* capplet)
+{
+	/* We enter in edit mode so change GUI to enter new name */
+	if (strcmp( gtk_button_get_label(button), "gtk-edit" ) == 0 )
+	{
+		gtk_widget_hide(capplet->button_network_toggle);
+		gtk_entry_set_text(capplet->entry_network, gtk_button_get_label(capplet->button_network_toggle) );
+		gtk_widget_show(capplet->entry_network);
+		gtk_button_set_label(GTK_BUTTON(button), "gtk-undo");
+	}
+	else
+	{
+		/* Leaving edit mode and set icon name to default value */
+		g_settings_reset(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
+		
+		gtk_button_set_label(GTK_BUTTON(capplet->button_network_toggle), _("NetWork"));
+		gtk_widget_show(capplet->button_network_toggle);
+		gtk_widget_hide(capplet->entry_network);
+		gtk_button_set_label(GTK_BUTTON(button), "gtk-edit");
+	}
+}
+
+/* Leave edit icon name mode and save new icon name for network icon */
+void
+on_activate_entry_network_cb(GtkWidget *entry, MateDICapplet* capplet)
+{
+    const gchar *icon_name = gtk_entry_get_text(GTK_ENTRY(entry));
+
+	g_settings_set_string(capplet->desktop_icons_settings, NETWORK_NAME_KEY, icon_name);
+ 
+    gtk_button_set_label(GTK_BUTTON(capplet->button_network_toggle), icon_name);
+
+	gtk_widget_show(capplet->button_network_toggle);
+	gtk_widget_hide(capplet->entry_network);
+	gtk_button_set_label(GTK_BUTTON(capplet->button_edit_network), "gtk-edit");
+}
+
+/* Get default icons names set by the user */
+void
+get_user_icons_names(MateDICapplet* capplet)
+{
+	/* Get computer icon name if set by the user */
+	g_autoptr(GVariant) *user_computer_name = g_settings_get_user_value(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
+	if (user_computer_name != NULL)
+	{
+		gchar *computer_name = g_settings_get_string(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
+		gtk_button_set_label(GTK_BUTTON(capplet->button_computer_toggle), computer_name);
+		g_free(computer_name);
+	}
+	g_variant_unref(user_computer_name);
+
+	/* Get home icon name if set by the user */
+	g_autoptr(GVariant) *user_home_name = g_settings_get_user_value(capplet->desktop_icons_settings, HOME_NAME_KEY);
+	if (user_home_name != NULL)
+	{
+		gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);
```

Comment:
You can use `g_autofree gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);`, then you shoud remove `g_free(home_name);`